scrapy/utils/url.py

```diff
@@ -133,3 +133,15 @@ def strip_url(url, strip_credentials=True, strip_default_port=True, origin_only=
         '' if origin_only else parsed_url.query,
         '' if strip_fragment else parsed_url.fragment
     ))
+
+
+def is_uri(text, protocols={'http', 'https', 'file'}):
+    regex = r'^(([a-zA-Z][-+.\w]*):)(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?'
+    pattern = re.compile(regex)
+    match = pattern.fullmatch(text)
```

Comment:
:lipstick: This would be simpler and the performance should be the same:

```suggestion
    match = re.fullmatch(pattern, text)
```